src/node/fetch/fetch.node.ts

```diff
@@ -148,26 +148,21 @@ export class NodeFetch implements IFetch, Interceptable, AuthCache {
     return new RequestFetchError('abort');
   }
 
-  private queryParameters(parameters?: Record<string, string>): string {
-    if (parameters && Object.keys(parameters).length) {
-      const definedParameters = Object.entries(parameters).reduce(
-        (result, [key, value]) => {
-          if (value === undefined) {
-            return result;
-          }
-
-          return {
-            ...result,
-            [key]: value,
-          };
-        },
-        {}
-      );
+  private queryParameters(parameters?: Record<string, string | string[]>): string {
+    if (parameters === undefined || Object.keys(parameters).length === 0) {
+      return '';
+    }
 
-      return '?' + new URLSearchParams(definedParameters).toString();
+    const params = new URLSearchParams();
+    for (const [key, param] of Object.entries(parameters)) {
+      if (typeof param === 'string') {
+        params.append(key, param);
+      } else {
```

Comment:
Hopefully we can rely on types.

Reply:
From how it is structured now this MultiMap comes from our variablesToHttpMap. I would also argue that if we can't trust the types at runtime we might as well not have them and do typeof checks everywhere, but I'd rather not do that 😅

Reply:
Well I am concerned, because of this line https://github.com/superfaceai/one-sdk-js/blob/3d95f6169d4ad669f5f297173bb953b61ef7b622/src/core/interpreter/map-interpreter.ts#L899

Reply:
It still goes through `variablesToHttpMap` (and `variablesToString` before these changes) before the request is dispatched to the fetch implementation, which ensures its type.

The cast is there because right now there is no way to infer the return type of `yield` based on input to `yield` - i.e.:
```
function* gen(): Generator<1 | 2, undefined, 'one' | 'two'> {
	const foo = yield 1;
	const bar = yield 2;

	return undefined;
}

const it = gen();
let next = it.next();
while (next.done !== true) {
	let stringified: 'one' | 'two';
	if (next.value === 1) {
		stringified = 'one';
	} else {
		stringified = 'two';
	}
	
	next = it.next(stringified);
}
```

There is no way to tell typescript that `foo: 'one'` and `bar: 'two'` even when we know this to be true. Same issue with the typecast - we are exploring `ObjectLiteralNode` which always returns `NonPrimitive`. I left the TODOs in because I wasn't sure whether we should typecheck, but thinking about it now I think it's not necessary. They should instead be SAFETY comments explaining why the cast is safe. This is an implementational detail and it should be tested, but it's not a reason to add a typecheck in a wholly separate class which may very well be in another package.

Reply:
Ok, thank you for explanation.